### PR TITLE
refactor transcribe-client

### PR DESCRIPTION
**Pull request by ppcssn**

### CODE REVIEW

File: transcribe/transcribe-client/cmd/transcribe.go [hunk 1]
```diff
@@ -29,115 +29,214 @@ import (
...
+	callBackFunc := func(resp *transcribepb.StreamingRecognizeResponse) {
+		if resp == nil {
 			return
 		}
 
-		if !response.Result.IsPartial && len(response.Result.Alternatives) > 0 {
-			resp = append(resp, *response) //nolint:govet // it is okay to copy
+		if resp.Error != nil {
+			logger.Error("msg", "recognition error", "error", resp.Error)
+		}
```
Comment (shahruk10) [inline transcribe/transcribe-client/cmd/transcribe.go:123]: Does the `resp` containing `Error` also contain `Result`? i.e. should we continue here

File: transcribe/transcribe-client/internal/client/client.go [hunk 2]
```diff
@@ -106,8 +166,16 @@ func (c *Client) ListModels(ctx context.Context) ([]*transcribepb.Model, error)
 // quickly and certainly not block.
 type RecognitionResponseHandler func(*transcribepb.StreamingRecognizeResponse)
 
+// StreamingRecognize wraps the bidirectional streaming API for performing
+// speech recognition. It sets up recognition using the given cfg. Data is
+// read from the given audio reader into a buffer and streamed to cubic
```
Comment (shahruk10) [inline transcribe/transcribe-client/internal/client/client.go:171]:
```suggestion
// read from the given audio reader into a buffer and streamed to Transcribe
```

**Review by shahruk10 — APPROVED** [review]
Looks good !

File: transcribe/transcribe-client/cmd/transcribe.go (hunk 1 above)
Comment (ppcssn) [reply transcribe/transcribe-client/cmd/transcribe.go:123]: it is none fatal error and according to [the docs](https://github.com/cobaltspeech/proto/blob/master/proto/cobaltspeech/cubic/v5/cubic.proto#L102-L104):

>If both `result` and `error` are present, the result is still valid.

so we should continue.

**Comment by ppcssn** [comment]
This need to check with customer docs, before merge in.

**Pull request merged**